src/js/main.js

```diff
@@ -6,38 +6,89 @@
     })//END OF REDIRECT
     .when ('/top-wishes', {
       templateUrl: 'partials/top-wishes.html',
-      controller: function ($http, $rootScope) {
+      controller: function ($http, $scope) {
         $http.get('http://wishcastr-staging.herokuapp.com/products/top.json')
         .then(function(response){
-            $rootScope.products = response.data;
+            $scope.products = response.data;
         })//END OF PROMISE
       }//end of controller
     })//END OF TOP-WISHES
     .when ('/user-wishes', {
       templateUrl: 'partials/user-wishes.html'
     })//END OF USER-WISHES
+
     .when ('/results', {
-      templateUrl: 'partials/results.html'
+      templateUrl: 'partials/results.html',
+      controller: 'Find',
+      controllerAs: 'results'
     })//END OF RESULTS
 
 
   })//END OF MODULE
-  .controller('Find', ['$http', '$scope', function($http, $scope){
-    var BASEURL = 'http://wishcastr-staging.herokuapp.com/products/';
-
-    $scope.query = "";
-    $scope.products = { };
-    console.log($scope.query);
-    $scope.search = function(){
-      $http.get(BASEURL+'search.json?query='+$scope.query)
+  .controller('Find', function($scope, Search){
+    $scope.data = {};
+    $scope.updateParam = function() {
+      Search.setParam($scope.data);
+    }
+    $scope.submitParam = function() {
+      Search.callApi()
       .then(function(response){
         $scope.products = response.data;
-      })//END PROMISE
-    }//END searchParam()
+      })//end promise
+    }//END SUBMITPARAM
+  }) //END CONTROLLER
+
+  .factory('Search', function($http){
+    var query = {};
+    var BASEURL = 'http://wishcastr-staging.herokuapp.com/products/search.json?query=';
+    var _param = '';
+    var _searchUrl = '';
+
+    var makeUrl = function(){
+      _param = _param.split(' ').join('+');
```

Comment:
The `$http` service will do this for you if you just pass it the query parameters as an object.